### ISSUE TITLE
Image Retraining: maybe reference retrain.py?

### ISSUE THREAD

**Issue by jmhodges** [tensorflow/hub#233]
Doc Link:
https://www.tensorflow.org/hub/tutorials/image_retraining

The Image Retraining document is nice because it talks about the behaviors you need around the retraining code. However, it doesn't go into much detail of how someone would produce that code to retrain the model. There's a 1.3K line `retain.py` linked from the tutorial takes a while to dig through. It might make things clearer if the tutorial referenced or displayed parts of `retrain.py` or something similar. 

For instance, other venues talk about "removing the output layer" in these models and it was unclear what that meant in the tensorflow context. The bottleneck discussion also mentioned "before the output layer". I was unsure if I'd have to take a Module's Graph and strip something off the end. I think showing the code involved might clear that up? 

Similarly, I think the bottleneck discussion might benefit from referencing the code that actually writes data down because, while it's probably obvious to others, I wasn't sure what all was going into those files. (I was also curious if there was some special ordering or collation being done.)

**Comment by akhorlin** [comment]
@arnoegw for reference

@jmhodges thank you for your feedback. We will consider it when we re-visit the structure of our colabs. Current image retraining tutorial was designed to be simple and easy to follow and retrain.py covers more advanced cases. We will look into potentially smoothing the transition between the two.

**Comment by arnoegw** [comment]
Hi @jmhodges, thanks for your thoughtful comments.

Please see [tensorflow.org/hub/common_signatures/images](https://www.tensorflow.org/hub/common_signatures/images) to alleviate the confusion about the origin of feature vectors ("bottleneck values"): Hub modules for extraction of image feature vectors already capture the correct piece of a pre-trained image classification model and can be used without any further adaptation. This is being taken care of when the modules are created by removing the logits layer that it was trained with.

I agree retrain.py in its current form is way too long to teach coding. It is also terribly out of sync with the change to TensorFlow 2.0 and the emphasis on using Keras as the high-level framework for expressing and training TensorFlow models.

For a fresher and much more concise example of how things will look like for TF Hub in TF2, please take a sneak peek at [examples/colab/tf2_image_retraining.ipynb](https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/tf2_image_retraining.ipynb), especially the section "Defining the model". As you can see, the module that extracts the vector of features from the image inputs fits right in, with no explicit adaptation required.

I think the overhaul of tutorials for TF2 will remove the current, lengthy retrain,py and resolve the issue that way.
